runhouse/resources/hardware/cluster.py

```diff
@@ -1237,6 +1229,15 @@ def _start_or_restart_helper(
             self._start_ray_workers(DEFAULT_RAY_PORT, env=self.default_env)
 
         self.put_resource(self.default_env)
+
+        if self._default_env.secrets:
```

Comment:
probably use `if default_env and default_env.secrets` instead, `self._default_env` can be None

Reply:
yep